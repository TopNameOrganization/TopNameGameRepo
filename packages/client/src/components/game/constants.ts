export const TileSize = 32;
export enum Tile {
  Empty = 0,
  Brick = 1,
  Concrete = 2,
  Stair = 3,
  Rope = 4,
  Bonus = 5,
  Enemy = 6,
  Player = 7,
  Out = 100,
}

export const VELOCITY = 200;
export enum Orientation {
  Left = 0,
  Right = 1,
}
export enum RunnerAction {
  Stay = 0,
  Fall = 1000,
  MoveLeft = 37,
  MoveUp = 38,
  MoveRight = 39,
  MoveDown = 40,
}

export enum AnimationPhase {
  Stay = 0,
  Run = 1,
  Climb = 2,
  Hang = 3,
  Fall = 4,
}
