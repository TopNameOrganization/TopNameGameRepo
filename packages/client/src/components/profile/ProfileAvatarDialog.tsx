import * as React from 'react';
import {AddCircleOutline} from "@mui/icons-material";
import {IconButton} from "@mui/material";
import DialogTitle from "@mui/material/DialogTitle";
import DialogContent from "@mui/material/DialogContent";
import DialogContentText from "@mui/material/DialogContentText";
import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import DialogActions from "@mui/material/DialogActions";
import Dialog from "@mui/material/Dialog";

export default function ProfileAvatarDialog(props: any) {
    const [open, setOpen] = React.useState<boolean>(false);

    const fileInput = React.createRef<HTMLInputElement>();

    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        const node: HTMLInputElement | null = fileInput.current
        if (node && node.files?.length) {
            console.log(`Selected file - ${node.files[0]}`)
            handleClose();
        }
    }

    const handleClickOpen = () => {
        setOpen(true);
    };

    const handleClose = () => {
        setOpen(false);
    };

    return (
        <>
            {
                props.isShowDialogButton &&
                <IconButton
                    aria-label="add"
                    sx={{
                        display: 'block',
                        position: 'absolute',
                        top: 0,
                        right: 0,
                        bottom: 0,
                        left: 0,
                        zIndex: 1,
                        backgroundColor: 'rgba(0,0,0, 0.1)',
                    }}
                    onClick={handleClickOpen}
                >
                    <AddCircleOutline sx={{fontSize: 50, color: 'white'}} />
                </IconButton>
            }
            <Dialog open={open} onClose={handleClose}>
                <DialogTitle>Change avatar</DialogTitle>
                <DialogContent>
                    <DialogContentText>
                        You can upload your avatar, a png, gif or jpeg file:
                    </DialogContentText>
                    <Box component="form" onSubmit={(e: React.FormEvent<HTMLFormElement>) => handleSubmit(e)} sx={{ mt: 1 }}>
                        <label htmlFor="avatar">
                            <input
                                name="avatar"
                                accept="image/png, image/gif, image/jpeg"
                                style={{ display: 'none' }}
                                id="avatar"
                                type="file"
                                required
                                ref={fileInput}
                            />
                            <Button component="span">
                                Upload file
                            </Button>
                        </label>
                        <DialogActions>
                            <Button onClick={handleClose}>Cancel</Button>
                            <Button type="submit" variant="contained">Save</Button>
                        </DialogActions>
                    </Box>
                </DialogContent>
            </Dialog>
        </>
    );
}
