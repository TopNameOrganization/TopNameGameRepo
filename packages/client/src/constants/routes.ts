export const ROUTES = {
  root: '/',
  home: '/home',
  game: '/game',
  login: '/login',
  signup: '/signup',
  profile: '/profile',
  leaderBoard: '/leader-board',
  forum: '/forum',
  forumCreate: '/forum/create',
  forumDetail: '/forum/:id',
}
