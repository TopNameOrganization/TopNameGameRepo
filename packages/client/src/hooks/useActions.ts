import {useDispatch} from "react-redux";
import {bindActionCreators} from "@reduxjs/toolkit";
import Actions from '../store/actions/index'

export const useActions = () => {
  const dispatch = useDispatch()
  return bindActionCreators(Actions, dispatch)
}
