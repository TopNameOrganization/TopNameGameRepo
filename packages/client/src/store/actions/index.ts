import * as UserAction from './userAction'
import * as UserProfileAction from './changeUserProfileAction'

export default {
  ...UserAction,
  ...UserProfileAction
}
