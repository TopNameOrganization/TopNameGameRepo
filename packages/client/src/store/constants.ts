import {UserState} from "./types";

export const initialState: UserState = {
  user: {
    id: null,
    first_name: '',
    second_name: '',
    display_name: '',
    login: '',
    email: '',
    phone: '',
    avatar: ''
  },
  loading: false,
  error: null
}
