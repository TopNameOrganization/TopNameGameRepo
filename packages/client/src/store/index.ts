import { rootReducer, RootState } from './reducers'
import { configureStore } from "@reduxjs/toolkit";

export const setupStore = (initialState?: RootState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState: initialState
  });
}
