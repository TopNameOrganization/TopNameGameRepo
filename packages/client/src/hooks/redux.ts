import {TypedUseSelectorHook, useDispatch, useSelector} from "react-redux";
import {AppDispatch, RootState} from "../store/reducers";
import {bindActionCreators} from "@reduxjs/toolkit";
import Actions from "../store/actions";

export const useAppDispatch = () => useDispatch<AppDispatch>()
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
export const useAppActions = () => bindActionCreators(Actions, useAppDispatch())
