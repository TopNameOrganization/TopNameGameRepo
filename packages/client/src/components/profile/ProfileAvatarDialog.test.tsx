import React from "react";
import * as renderer from 'react-test-renderer';
import ProfileAvatarDialog from './ProfileAvatarDialog';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import * as appHooks from "../../hooks/redux";
import { Provider } from "react-redux";
import { userInitialState } from "../../store/constants";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const useAppSelectorMock = jest.spyOn(appHooks, 'useAppSelector');

describe('ProfileAvatarDialog', () => {
  it('should renders profile correctly', () => {
    const isShow = false
    useAppSelectorMock.mockReturnValue(userInitialState)
    const store = mockStore(userInitialState)
    const tree = renderer
        .create(
            <Provider store={store}>
                <ProfileAvatarDialog isShowDialogButton={isShow} />
            </Provider>)
        .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
