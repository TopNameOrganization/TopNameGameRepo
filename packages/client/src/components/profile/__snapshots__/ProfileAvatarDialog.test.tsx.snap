// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileAvatarDialog should renders profile correctly 1`] = `null`;
